evil-ex.el

```diff
@@ -458,6 +458,11 @@ in case of incomplete or unknown commands."
 (defun evil-ex-argument-completion-at-point ()
   (let ((context (evil-ex-syntactic-context (1- (point)))))
     (when (memq 'argument context)
+      ;; if it's an autoload, load the function so ex completion can be provided
```

Comment:
I'd extend this comment by an explanation how this situation happens in the first place as it's not immediately obvious.  Something like a one-sentence version of your original post would work.